src/common/config.ts

```diff
@@ -280,24 +319,38 @@ function parseCliConfig(args: string[]): CliOptions {
     return parsed;
 }
 
-export function warnAboutDeprecatedCliArgs(
-    args: CliOptions &
-        UserConfig & {
-            _?: string[];
-        },
-    warn: (msg: string) => void
-): void {
+export function warnAboutDeprecatedOrUnknownCliArgs(args: object, warn: (msg: string) => void): void {
     let usedDeprecatedArgument = false;
+    let usedInvalidArgument = false;
+
+    const knownArgs = args as UserConfig & CliOptions;
     // the first position argument should be used
     // instead of --connectionString, as it's how the mongosh works.
-    if (args.connectionString) {
+    if (knownArgs.connectionString) {
```

Comment:
```suggestion
export function warnAboutDeprecatedOrUnknownCliArgs(args: Record<string, unknown>, warn: (msg: string) => void): void {
let usedDeprecatedArgument = false;
    let usedInvalidArgument = false;

// the first position argument should be used
// instead of --connectionString, as it's how the mongosh works.
    if (args.connectionString) {
```
I think a Record might be more appropriate here compared to object as its more narrower?